### ISSUE TITLE
Yay should respect ~ for $HOME in config file

### ISSUE THREAD

**Issue by bulletmark** [Jguer/yay#639]
Version is yay v8.998 - libalpm v11.0.1

Yay is one of the few programs which does not respect a '`~`' as an alias for `$HOME` for file paths specified in `~/.config/yay/config.json`. So, for example, you have to hard code the path `/home/mark/.cache/yay/` for `buildDir` instead of being able to specify `~/.cache/yay/`. When I copy my config files to other users (e.g. on other machines) I have to remember to manually edit yay's config for the new path.

BTW, a separate issue I know but `buildDir` should default to the above, like pacaur does.

**Comment by Morganamilo** [comment]
`~` is a shell thing and I wouldn't say most programs support it. Just most programs which happen to have their config files written in bash. For example pacman does not support this.

I get that the current way is a little annoying, right now I'm torn on two solutions:

1. Have builddir be an empty string, this causes it to resolve to the current users cache every time. The user can then override it if they wish.
2. Support expanding environment variables. This is basically what you want but `$HOME` instead of `~`.

>BTW, a separate issue I know but buildDir should default to the above, like pacaur does.

I have no idea what youre talking about here.

**Comment by bulletmark** [comment]
I wouldn't say tilde parsing in paths is a "shell thing". E.g. Python has always had a native function to parse it (see https://docs.python.org/3/library/os.path.html#os.path.expanduser and https://docs.python.org/3/library/pathlib.html#pathlib.Path.expanduser where there is no mention of being a shell like feature) and I presume many other languages have that also (including go?). Note also that tilde parsing is not just about replacing $HOME.

Solution 2 is what trizen provides, and possibly other helpers.

> I have no idea what youre talking about here.

I was saying that I think it best that yay by default use `~/.cache/yay` for the buildDir because when I started using yay it defaulted to `/tmp` but I just deleted my config.json now and I see that the fresh one yay now creates DOES default to `/home/me/.cache/yay`. It seems like this was changed? Note my config.json was missing a huge number of config values so it looks like it was changed quite some time ago. Either way, that change was what I was suggesting, although it would be nice to allow `~` in paths as I suggest above.

**Comment by Morganamilo** [comment]
`~` has always been a shell thing to me at least. Sure I see it supported in other programs occasionally. openssh being one that I know of.

And of course python has it in its stdlib. Python seems to have every conceivable function in there. But no, go does not. If it did I might implement it, but I for sure am not writing my own function to do this, especially when https://golang.org/pkg/os/#ExpandEnv exists.

So ExpandEnv will probably be the solution I go with.

>yay now creates DOES default to /home/me/.cache/yay. It seems like this was changed?

You're about 7 months late on that realisation :P

**Comment by bulletmark** [comment]
> You're about 7 months late on that realisation

"Late" and "realisation" are hardly appropriate words though.  When I started using yay long ago I immediately found that it unfortunately defaulted the build dir to `/tmp` like yaourt does rather than under the better `~/.cache` option like pacaur does. Since I came from pacaur (like probably most new yay users) that was the first thing I changed and it is not as if yay is suddenly going to pop up and tell me about it on the day that issue was "fixed" is it?

Back to the subject matter of this issue, as another example, trizen allows for avoiding hard coding your absolute home dir as seen at https://github.com/trizen/trizen/issues/75.
